test/Microsoft.Net.Http.Headers.Tests/NameValueHeaderValueTest.cs

```diff
@@ -575,6 +575,83 @@ public void TryParseStrictList_WithSomeInvlaidValues_ReturnsFalse()
             Assert.False(NameValueHeaderValue.TryParseStrictList(inputs, out results));
         }
 
+        [Theory]
+        [InlineData("value", "value")]
+        [InlineData("\"value\"", "value")]
+        [InlineData("\"hello\\\\\"", "hello\\")]
+        [InlineData("\"hello\\\"\"", "hello\"")]
+        [InlineData("\"hello\\\"foo\\\\bar\\\\baz\\\\\"", "hello\"foo\\bar\\baz\\")]
+        [InlineData("\"quoted value\"", "quoted value")]
+        [InlineData("\"quoted\\\"valuewithquote\"", "quoted\"valuewithquote")]
+        public void GetUnescapedValue_ReturnsExpectedValue(string input, string expected)
+        {
+            var header = new NameValueHeaderValue("test", input);
+
+            var actual = header.GetUnescapedValue();
+
+            Assert.Equal(expected, actual);
+        }
+
+        [Theory]
+        [InlineData("\"hello\\\"")]
+        // See https://github.com/aspnet/HttpAbstractions/issues/923 for why invalid escape characters do not throw.
```

Comment:
Suggest clarifying the comment. Not clear you're talking about the `NameValueHeaderValue` constructor accepting invalid arguments.